### DISCUSSION TITLE
Loading a custom measured BRDF

### DISCUSSION THREAD

**Discussion by gioarma** [mitsuba-renderer/mitsuba3#906] [Q&A] [answered]
In my lab I will soon have the possibility of measuring BRDF of materials, and it would be great to be able to load them on Mitsuba for rendering.
However, I'm struggling to understand how to convert my measured tabular data in into a .bsdf file that Mitsuba can import.
I took a look at the .bsdf files in the material database (http://rgl.epfl.ch/materials), but I saw that they are in a binary format. I saw they also provide tabular raw data, but those are not the ones imported by the software.

Can anyone tell me how sould I treat my raw bsdf data to obtain a .bsdf file readable by Mitsuba?

Thank you in advance

**Comment by njroussel** [comment, marked as answer]
Hi @gioarma 

IIRC this format is somewhat tailored to the [method presented in the paper](http://rgl.epfl.ch/publications/Dupuy2018Adaptive). 

The repository for that project can be found [here](https://github.com/rgl-epfl/brdf-loader). Specifically, you'll want to have a look at [this file](https://github.com/rgl-epfl/brdf-loader/blob/master/python/visualize.py) which holds the routines to read write these `.bsdf` files from raw tabulated data.

I hope this helps :smile:!

**Reply by gioarma** [reply]
Hello @njroussel !
Thank you for your answer! I was indeed able to import the data using the `read_tensor` function. So now I can arrange my data in the same way as the input file, and then export them to the correct format with `write_tensor`

Thanks again!
